internal/printer/print-to-js.go

```diff
@@ -907,25 +915,45 @@ func generateEndSlotIndexes(nestedSlotChildren []*NestedSlotChild) map[int]bool
 	return endSlotIndexes
 }
 
-func mergeDefaultSlotsAndUpdateIndexes(nestedSlotChildren *[]*NestedSlotChild, endSlotIndexes map[int]bool) {
-	defaultSlot := &NestedSlotChild{SlotProp: `"default"`, Children: []*Node{}}
-	mergedSlotChildren := make([]*NestedSlotChild, 0)
-	numberOfMergedSlotsInSlotChain := 0
+func mergeDefaultSlotsAndUpdateIndexes(nestedSlotChildren *[]*NestedSlotChild, endSlotIndexes *map[int]bool) {
+	defaultSlot := &NestedSlotChild{SlotProp: DEFAULT_SLOT_PROP, Children: []*Node{}}
+	updatedNestedSlotChildren := make([]*NestedSlotChild, 0)
+	updatedEndSlotIndexes := make(map[int]bool)
 
 	for i, nestedSlot := range *nestedSlotChildren {
+		var isDefault bool
 		if isDefaultSlot(nestedSlot) {
+			isDefault = true
 			defaultSlot.Children = append(defaultSlot.Children, nestedSlot.Children...)
-			numberOfMergedSlotsInSlotChain++
 		} else {
-			mergedSlotChildren = append(mergedSlotChildren, nestedSlot)
-		}
-		if shouldMergeDefaultSlot(endSlotIndexes, i, defaultSlot) {
-			resetEndSlotIndexes(endSlotIndexes, i, &numberOfMergedSlotsInSlotChain)
-			mergedSlotChildren = append(mergedSlotChildren, defaultSlot)
-			defaultSlot = &NestedSlotChild{SlotProp: `"default"`, Children: []*Node{}}
+			updatedNestedSlotChildren = append(updatedNestedSlotChildren, nestedSlot)
+		}
+
+		// we reached the end of a slot chain
+		if (*endSlotIndexes)[i] {
+			// free up memory, this information is now outdated
+			// the updated information is stored in updatedEndSlotIndexes
+			delete(*endSlotIndexes, i)
+
+			if len(defaultSlot.Children) > 0 {
+				// if the last element in a slot chain is a default slot
+				// let's add it to the updated nested slot children
+				updatedEndSlotIndexes[len(updatedNestedSlotChildren)] = true
+				updatedNestedSlotChildren = append(updatedNestedSlotChildren, defaultSlot)
+				defaultSlot = &NestedSlotChild{SlotProp: DEFAULT_SLOT_PROP, Children: []*Node{}}
+			} else if !isDefault {
+				// if it's not a default slot, we just need to set the updated end slot indexes
+				// we already added it in the previous iteration
+				updatedEndSlotIndexes[len(updatedNestedSlotChildren)-1] = true
```

Comment:
Do you know if this operation is safe? Are we sure that `updatedNestedSlotChildren` has at least one element at this point? Is the lenght of `updatedEndSlotIndexes` high enough to store this new information? 

Sorry, I have minimal context so my questions my sound dumb